azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/HPFlowServlet.java

```diff
@@ -0,0 +1,129 @@
+  }
+
+  @Override
+  public void init(final ServletConfig config) throws ServletException {
+    super.init(config);
+    final AzkabanWebServer server = getApplication();
+    this.converterUtils = server.getConverterUtils();
+    this.hpFlowService = server.getHPFlowService();
+  }
+
+  @Override
+  protected void handleGet(final HttpServletRequest req,
+      final HttpServletResponse resp, final Session session)
+    throws ServletException, IOException {
+
+  }
+
+  @Override
+  protected void handlePost(final HttpServletRequest req,
+      final HttpServletResponse resp, final Session session)
+      throws ServletException, IOException {
+    handleAddHPFlows(req, resp, session);
+  }
+
+  /**
+   * This method takes in a CSV of high priority flows and adds them to metadata.
+   * @param req
+   * @param resp
+   * @param session
+   * @throws ServletException
+   * @throws IOException
+   */
+  protected void handleAddHPFlows(final HttpServletRequest req,
+      final HttpServletResponse resp, final Session session)
+      throws ServletException, IOException {
+    try {
+      final String jsonPayLoad = HttpRequestUtils.getBody(req);
+      final HPFlowDTO hpFlowDTO = this.converterUtils.convertToDTO(jsonPayLoad, HPFlowDTO.class);
+      // If there are no flow IDs in the list return with error
+      if (hpFlowDTO.getFlowIds().isEmpty()) {
+        log.error("There are no flow IDs provided");
+        throw new ImageMgmtValidationException(ErrorCode.BAD_REQUEST,
+            "Required field flow IDs is empty");
+      }
+      // Check for required permission to invoke the API
+      final User user = session.getUser();
+      if (!hasHPFlowManagementPermission(user)) {
+        log.debug(String.format("Invalid permission to access High Priority "
+            + "flows for user: %s", user.getUserId()));
+        throw new ImageMgmtInvalidPermissionException(ErrorCode.FORBIDDEN,
+            "Invalid permission to manage high priority flows");
+      }
+      // Set the user who invoked the API
+      hpFlowDTO.setCreatedBy(user.getUserId());
+      // Add the HP flow metadata
+      this.hpFlowService.addHPFlows(hpFlowDTO);
+      // Prepare to send response
+      resp.setStatus(HttpStatus.SC_CREATED);
+      sendResponse(resp, HttpServletResponse.SC_CREATED, new HashMap<>());
+    } catch (final ImageMgmtException e) {
+      log.error("Exception while adding high priority flows.", e);
+      sendErrorResponse(resp, e.getErrorCode().getCode(), e.getMessage());
+    } catch (final Exception e) {
+      log.error("Exception while adding high priority flows", e);
+      sendErrorResponse(resp, HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+          "Exception while adding high priority flows. " + e.getMessage());
+    }
+  }
+
+  /**
+   * Checks if the provided user has access to manage high priority flows.
+   * @param user
+   * @return
+   */
+  private boolean hasHPFlowManagementPermission(final User user) {
```

Comment:
I might be incorrect as I am going from memory, but wouldn't a direct call to the permission manager provide the same functionality?

Reply:
This API checks if the user is Azakaban admin or not, if yes, then PermissionManager is not called.

Reply:
You're correct. I was miss-remembering. Thank you.